### DISCUSSION TITLE
which translations to maintain?

### DISCUSSION THREAD

**Discussion by 0foxes** [dogecoin/dogecoin#3460] [General]
From a quick look through src/qt/locale, many languages we currently have "support" for are 50-90% untranslated. I have several general questions:
- What is the maintenance burden of keeping untranslated languages around? 
- Is it worth keeping "translations" around if most strings are untranslated?
- Do we want translators to continue using the Qt Linguist/editing XML workflow?
- 
- Are there any countries where a) english isn't prevalent b) there are many dogecoin users c) the translation is very incomplete?
(I have been checking using https://what-is-dogecoin.com/nodes/ but don't know how accurate it is)

**Comment by daank-c** [comment]
> Is it worth keeping "translations" around if most strings are untranslated?

I think so. When new Core users come online, they'll notice their language isn't fully supported and potentially come to help. This maybe isn't ideal, but it might do better at encouraging participation than not having the language supported at all.

> Do we want translators to continue using the Qt Linguist/editing XML workflow?

I think it works okay but it's not super accessible, and it's my belief that accessibility is at the root of participation. On that note, I was thinking about this recently and decided to reach out to Transifex to see what it would cost to use their service. This is the response I was given:

```
Thanks for taking an interest in Transifex!

Regarding your open-source inquiry, to 
create an open-source project, the project
should not have funding, revenue, or 
commercialization model and you will be 
asked to provide the following:

- A link to your project's code repository.
- The repository must contain a LICENSE 
  file stating your project's license type.

Here is more information on our open-source
projects:
https://help.transifex.com/en/articles/6236788-open-source-projects.
```

**Reply by daank-c** [reply]
Hmmm. I'm wondering if maybe there should/could be something that warns a user when they select a language that isn't 100% complete.

Like, if `x` language is selected then an alert comes up saying "oof. it seems some translations for this language are unfinished. Would you like to continue? (y/n)"

This way they know what to expect upon selecting that language, and might still encourage participation. (maybe this is bad UX though.)

**Reply by patricklodder** [reply]
There are exactly 3 languages that are 100% translated (but that doesn't mean 100% correct either): `ko_KR`, `nl` and `zh_CN`. Everything else has at least 1 (`it`) or more unfinished translations.

**Reply by daank-c** [reply]
I would like to help improve that, as I believe that internationalization--and accessibility features in general--are enablers of participation (on the network, in development, etc.).

Is there currently a threshold for completeness (in policy, or code), that dictates when Core drops support for a language?

**Reply by patricklodder** [reply]
No, but we can make a policy. The dialects (not primary languages) we dropped with 1.14.7 were deprecated with 1.14.6, because they were copied over from Bitcoin and had never seen any work on this repository. There are also primary languages that are completely unmaintained and never gotten any attention at all... but these are like a trigger to do something - i.e. see `ar` translation.

If we're going to set up Transifex, we can start with what we have, mark (dialect) translations that don't get maintained as candidate to be removed, and monitor reactions. Sometimes announcing something as unmaintained helps getting translations done, and it makes it easy to contribute, as it would be open.

**Reply by 0foxes** [reply]
I agree, it would be good if we could announce somewhere (GitHub, Reddit, Twitter?) that we are looking for maintainers of [list of languages with little translation work done] translations. 

What dialects would you suggest further removing support for? It seems we have the following dialects (that we maintain multiple translations of):
en/en_GB, pt_BR/pt_PT, sr/sr@latin, zh_CN/zh_TW, fr/fr_CA

I've only used Qt Linguist and Crowdin, but I'm not opposed to Transifex if it gets more people involved.

**Reply by 0foxes** [reply]
Also @patricklodder how are you checking translation completeness? Just for my personal curiosity.

**Reply by patricklodder** [reply]
> What dialects would you suggest further removing support for?

`af_ZA` - I missed that one last time. It's never been touched so it would be easier to get one good `af` translation first, and differentiate after.

> I've only used Qt Linguist and Crowdin

The upside of Qt Linguist is full decentralization and it's open source. We'd need a real guide for that though - not the ad-hoc crap I'm doing.

> Also how are you checking translation completeness?

```bash
for f in $(find src/qt/locale -name "bitcoin_*.ts" | sort); do echo -n "$f: "; grep unfinished $f |wc -l; done
```

**Reply by daank-c** [reply]
I don't mind going through and removing all of the erroneous "unfinished" tags that have been left behind over the years. This would give me something to do, and help us get a more accurate reading on completeness.

**Reply by patricklodder** [reply]
`"unfinished"` tags aren't erroneous, so please don't delete those. What you probably mean is `"vanished"`. We keep those on purpose, for situations where just a few words changed (most of 'em) and there is a reference in there what the previous translation was.

**Reply by daank-c** [reply]
Oh okay, my mistake. I figured the translations with "unfinished" tags, that actually have translations, were left behind unintentionally.

For example from `bitcoin_af.ts`:

```
Line 957-960

    <message>
        <source>Rescanning...</source>
        <translation type="unfinished">Word herskandeer...</translation>
    </message>
 ```

**Reply by patricklodder** [reply]
2 things are possible:

1. The translator left it as unfinished
2. Qt `lupdate` did a fuzzy match on the text - eg when it moved from one module to another - and it needs to be confirmed.

It just means that it needs to be edited by someone that speaks the language. I think a golden rule to translations is: let's not edit anything for which we don't speak the language, unless it's scripted by Qt Translations tools.

**Reply by daank-c** [reply]
Noted. Thank you for the clarification.

**Comment by patricklodder** [comment]
> What is the maintenance burden of keeping untranslated languages around?

None, because updates are scripted. The reason for removing untranslated dialects is because it is confusing.

> Is it worth keeping "translations" around if most strings are untranslated?

I'm neutral on this.

> Do we want translators to continue using the Qt Linguist/editing XML workflow?

I don't mind moving to transifex if translators believe that that gives a better experience.

> Are there any countries where a) english isn't prevalent b) there are many dogecoin users c) the translation is very incomplete?

I feel that we cannot include (b) here, because maybe there aren't many users because there is no complete translation. So I think that it's better to make decisions based on how large the total population is we can reach with a translation.

**Reply by 0foxes** [reply]
You raise a good point about (b). From a quick check, I think some countries are 
- India (Hindi, almost completely untranslated)
- Pakistan (Urdu, completely untranslated)
- Bangladesh (Bengali, no file)
- Ethiopia (Amharic, no file)
- Egypt (Arabic, PR at https://github.com/dogecoin/dogecoin/pull/3439)

About what I expected, countries with sizable populations that interact less with the english-speaking internet.

Large countries that already have a mostly (>80%) finished translation:
- China (Chinese)
- United States, Nigeria, Philippines (English)
- Indonesia (Indonesian)
- Brazil (Portuguese, see [prior discussion](https://github.com/dogecoin/dogecoin/pull/3029))
- Russia (Russian)
- Mexico (Spanish)
- Japan (Japanese)
- DR Congo (French)

**Reply by 0foxes** [reply]
Out of the 10 most spoken languages in the world (arbitrary bar), we are missing Hindi (3), Arabic (6), Bengali (7), and Urdu (10).
